src/storage/src/hummock/compactor/shared_buffer_compact.rs

```diff
@@ -532,7 +531,6 @@ impl SharedBufferCompactRunner {
                 key_range,
                 cache_policy: CachePolicy::Fill(CacheContext::Default),
                 gc_delete_keys: GC_DELETE_KEYS_FOR_FLUSH,
-                watermark: GC_WATERMARK_FOR_FLUSH,
```

Comment:
For shared buffer compact, if this watermark is not specified, will a value written in epoch non-checkpoint barrier be stored in compaction if a new value is written in a later epoch of checkpoint barrier?

Reply:
Fixed.